diff --git i/Makefile.defs w/Makefile.defs
index 6240b3be..b50c0aa4 100644
--- i/Makefile.defs
+++ w/Makefile.defs
@@ -930,6 +930,9 @@ endif
 
 CMAKE_KOVARS = $(CMAKE_DIR)/koreader_vars.cmake
 CMAKE_FLAGS += -DCMAKE_KOVARS='$(abspath $(CMAKE_KOVARS))'
+ifdef LUAJIT_HOST_CC32
+CMAKE_FLAGS += -DLUAJIT_HOST_CC32=$(LUAJIT_HOST_CC32)
+endif
 
 define cmake_koreader_vars
 
@@ -1020,8 +1023,8 @@ cpp = [CCACHE, '$(strip $(REALCXX))']
 # cpp_ld
 ar = '$(strip $(AR))'
 nm = '$(strip $(NM))'
-pkgconfig = 'pkg-config'
-pkg-config = 'pkg-config'
+pkgconfig = '$(if $(MESON_PKGCONFIG),$(MESON_PKGCONFIG),pkg-config)'
+pkg-config = '$(if $(MESON_PKGCONFIG),$(MESON_PKGCONFIG),pkg-config)'
 ranlib = '$(strip $(RANLIB))'
 strip = 'true'
 
@@ -1047,7 +1050,7 @@ MESON_HOST_TOOLCHAIN = $(CMAKE_DIR)/meson_host_toolchain.ini
 MESON_CROSS_SYSTEM = $(if $(ANDROID),android,linux)
 ifneq (,$(filter aarch64-%,$(TARGET_MACHINE)))
   MESON_CROSS_CPU_FAMILY = aarch64
-else ifneq (,$(filter arm-% armv7a-%,$(TARGET_MACHINE)))
+else ifneq (,$(filter arm-% armv7a-% armv7l-%,$(TARGET_MACHINE)))
   MESON_CROSS_CPU_FAMILY = arm
 else ifneq (,$(filter i686-%,$(TARGET_MACHINE)))
   MESON_CROSS_CPU_FAMILY = x86
