diff --git i/Makefile.defs w/Makefile.defs
index 6ff1aeca..278bd603 100644
--- i/Makefile.defs
+++ w/Makefile.defs
@@ -801,7 +801,7 @@ CMAKE_FLAGS += -DCMAKE_TOOLCHAIN_FILE='$(abspath $(CMAKE_TCF))'
 # (Cf. https://github.com/taka-no-me/android-cmake)
 ifneq (,$(filter aarch64-%,$(CHOST)))
   CMAKE_SYSTEM_PROCESSOR = aarch64
-else ifneq (,$(filter arm-% armv7a-%,$(CHOST)))
+else ifneq (,$(filter arm-% armv7a-% armv7l-%,$(CHOST)))
   CMAKE_SYSTEM_PROCESSOR = arm
 else ifneq (,$(filter i686-%,$(CHOST)))
   CMAKE_SYSTEM_PROCESSOR = i686
@@ -921,6 +921,9 @@ endef
 
 CMAKE_KOVARS = $(CMAKE_DIR)/koreader_vars.cmake
 CMAKE_FLAGS += -DCMAKE_KOVARS='$(abspath $(CMAKE_KOVARS))'
+ifdef LUAJIT_HOST_CC32
+CMAKE_FLAGS += -DLUAJIT_HOST_CC32=$(LUAJIT_HOST_CC32)
+endif
 
 define cmake_koreader_vars
 
@@ -1012,8 +1015,8 @@ cpp = [CCACHE, '$(strip $(REALCXX))']
 # cpp_ld
 ar = '$(strip $(AR))'
 nm = '$(strip $(NM))'
-pkgconfig = 'pkg-config'
-pkg-config = 'pkg-config'
+pkgconfig = '$(if $(MESON_PKGCONFIG),$(MESON_PKGCONFIG),pkg-config)'
+pkg-config = '$(if $(MESON_PKGCONFIG),$(MESON_PKGCONFIG),pkg-config)'
 ranlib = '$(strip $(RANLIB))'
 strip = 'true'
 
@@ -1039,7 +1042,7 @@ MESON_HOST_TOOLCHAIN = $(CMAKE_DIR)/meson_host_toolchain.ini
 MESON_CROSS_SYSTEM = $(if $(ANDROID),android,linux)
 ifneq (,$(filter aarch64-%,$(CHOST)))
   MESON_CROSS_CPU_FAMILY = aarch64
-else ifneq (,$(filter arm-% armv7a-%,$(CHOST)))
+else ifneq (,$(filter arm-% armv7a-% armv7l-%,$(CHOST)))
   MESON_CROSS_CPU_FAMILY = arm
 else ifneq (,$(filter i686-%,$(CHOST)))
   MESON_CROSS_CPU_FAMILY = x86
